SourceX/controls/controller.cpp

```diff
@@ -36,7 +127,7 @@ ControllerButtonEvent ToControllerButtonEvent(const SDL_Event &event)
 	}
 #endif
 
-	const Joystick *joystick = Joystick::Get(event);
+	Joystick *const joystick = Joystick::Get(event);

```

Comment:
Why? You don't seem to be modifying `joystick`?

Reply:
Yeah, It could be reverted, I just wanted to keep the same order as in controller. Do you prefer if I rollback?

Reply:
They mean different things:

1. `const Joystick *joystick`: a non-const pointer to a const `Joystick`
2. `Joystick *const joystick`: a const pointer to a non-const `Joystick`

In case of 1, you can't modify the joystick, though you can assign another `const Joystick *` to the `joystick` variable.

In case of 2, you can modify the joystick but not reassign the variable.

Reply:
The most const-correct version here would be

```c++
const Joystick *const joystick = Joystick::Get(event)
```

Reply:
Right! Done.